src/main/java/com/hermesworld/ais/galapagos/topics/controller/TopicController.java

```diff
@@ -127,13 +130,19 @@ public void addProducerToTopic(@PathVariable String environmentId, @PathVariable
         }
 
         try {
-            topicService.removeProducerFromTopic(envId, topicName, producerApplicationId).get();
+            topicService.removeTopicProducer(envId, topicName, producerApplicationId).get();
             return new ResponseEntity<>(HttpStatus.NO_CONTENT);
 
         }
         catch (ExecutionException e) {
             throw handleExecutionException(e);
         }
+        catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
It's fine to add a `return null` here to satisfy the compiler. This command immediately exits the method.